docs/source/architecture/slashing.rst

```diff
@@ -0,0 +1,37 @@
+The Slashing Protocol
+=====================
+
+TBD
+
+
+Violations
+----------
+
+TBD
+
+
+Calculating the slashing penalty
+--------------------------------
+
+TBD (https://github.com/nucypher/nucypher/issues/803)
+
+
+How slashing affects stake
+--------------------------
+
+The punishment is to reduce the number of tokens that belongs to the offending staker.
+In this case, the main task is not to violate the logic of locking tokens.
+The whole stake consists of several parts:
+
+* tokens which the staker can withdraw at any moment
+* tokens locked for a specific period
+
+Since a staker can divide a stake (extend a part of a stake), as well as lock new tokens, then the total stake is represented as the sum of all the sub stakes active at a particular period. Each sub stake has a beginning and duration, which is reduced only upon confirmation of the activity. Staker slashed on the principle: first a sub stake, which will end earlier. Therefore, in the first place, the not-locked part of tokens belonging to the staker is slashed. After that, if necessary, sub stakes are adjusted: finds the shortest sub stake and decreases by the required amount. If this is not enough, then the next short sub stake is searched for, and so on. A sub stake that begins in the next period is checked separately.
```

Comment:
```suggestion
Since a staker can divide a stake (extend a part of a stake), as well as lock new tokens, then the total stake is represented as the sum of all the sub stakes active at a particular period. Each sub stake has a beginning and duration, which is reduced only upon confirmation of the activity. Staker slashed on the principle: first a sub stake, which will end earlier. Therefore, in the first place, the unlocked part of tokens belonging to the staker is slashed. After that, if necessary, sub stakes are adjusted: the shortest sub stake is found and decreased by the required amount; if this is not enough, then the next short sub stake is searched for, and so on. A sub stake that begins in the next period is checked separately.
```